### DISCUSSION TITLE
Why do all widgets have an ```as_group()``` method?

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#793] [Q&A]
Why do all widgets have an ```as_group()``` method? 
For example ToggleButton, Frame, etc. all have this method. Seems like most (if not all?) of these widgets return a ```None``` with ```as_group()```. 

The fact that they do have the method might lead one to think you could somehow turn the widgets into a GroupExt

**Comment by MoAlyousef** [comment]
It helps with type erased widgets. It's safe in FLTK to cast anything back to an Fl_Widget since everying derives from it, and it's what you get in all callbacks, as_group allows similar things for group widgets, and to get a valid Group widget from a Widget if the widget were derived from Group.
For example in Rust, if you have a collection of type erased widgets (Widget or `Box<dyn WidgetExt>`, if as_group returns Some, you can also acquire its children Widgets.
